Task6_MongoDB.txt

```diff
@@ -0,0 +1,262 @@
+       Task6. Aggregating Airlines Collection
+
+mongoimport -d frontcamp -c airlines --type csv --headerline --file airlines.csv
+
+Query:    db.airlines.count()
+Result:   186648
+ 
+
+
+
+1. How many records does each airline class have? Use $project to show result as 
+   { class: "Z", total: 999 }
+
+Query:  db.airlines.aggregate([
+    {
+        $group: {
+            _id: "$class",
+            total: {
+                $sum: 1
+            }
+        }
+    },
+    {
+        $project: {
+            _id: 0,
+            class: "$_id",
+            total: "$total"
+        }
+    }
+])
+
+Result: { "class" : "F", "total" : 140343 }
+	{ "class" : "L", "total" : 23123 }
+	{ "class" : "P", "total" : 5683 }
+	{ "class" : "G", "total" : 17499 }
+
+---------------------------------------------------------------------------
+2. What are the top 3 destination cities outside of the United States (destCountry field, not
+   included) with the highest average passengers count? Show result as 
+   { "avgPassengers" : 2312.380, "city" : "Minsk, Belarus" }
+
+Query:   db.airlines.aggregate([
+    {
+        $match: {
+            destCountry: {
+                $ne: "United States"
+            }
+        }
+    },
+    {
+       $group: {
+            _id: "$destCity",
+            avgPassengers: {$avg: "$passengers"}
+       }
+    },
+    {
+        $sort: {
+            avgPassengers: -1
+        }
+    },
+    {
+        $project: {
+            _id: 0,
+            avgPassengers: "$avgPassengers",
+            city: "$_id"
+        }
+    },
+    {
+        $limit: 3
+    }
+])
+
+Result: { "avgPassengers" : 8052.380952380952, "city" : "Abu Dhabi, United Arab Emirates" }
+	{ "avgPassengers" : 7176.596638655462, "city" : "Dubai, United Arab Emirates" }
+	{ "avgPassengers" : 7103.333333333333, "city" : "Guangzhou, China" }
+
+---------------------------------------------------------------------------
+3. Which carriers provide flights to Latvia (destCountry)? Show result as one document 
+   { "_id" : "Latvia", "carriers" : [ "carrier1", " carrier2", �] }
+
+Query:  db.airlines.aggregate([
+    {
+        $match: {
+            destCountry: "Latvia", 
+        }
+    },
+    {
+        $group: {
+            _id: "$destCountry",
+            carriers: {$addToSet: "$carrier"}
+        }
+    }
+])
+
+Result: { "_id" : "Latvia", "carriers" : [ "Uzbekistan Airways", "Blue Jet SP Z o o", "JetClub AG" ] }
+
+---------------------------------------------------------------------------
+4. What are the carriers which flue the most number of passengers from the United State to either
+   Greece, Italy or Spain? Find top 10 carriers, but provide the last 7 carriers (do not include the
+   first 3). Show result as { "_id" : "<carrier>", "total" : 999}
+
+Query:   db.airlines.aggregate([

```

Comment:
How do you manage `Greece, Italy or Spain` condition in your query?

Reply:
Done